src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -371,6 +374,96 @@ def image_mask_iterator():
     return nwbfile
 
 
+def add_fluorescence(segmentation_extractor: SegmentationExtractor, nwbfile: NWBFile, metadata: dict) -> NWBFile:
+    """
+    Adds the fluorescence specificied by the metadata to the nwb file.
+    The fluorescence that is added is the one located in metadata["Ophys"]["Fluorescence"]
+
+    Parameters
+    ----------
+    nwbfile : NWBFile
+        An previously defined -in memory- NWBFile.
+    metadata : dict
+        The metadata in the nwb conversion tools format.
+
+    Returns
+    -------
+    NWBFile
+        The nwbfile passed as an input with the fluorescence added.
+    """
+
+    # Set the defaults and required infrastructure
+    metadata_copy = deepcopy(metadata)
+    default_metadata = get_default_ophys_metadata()
+    metadata_copy = dict_deep_update(default_metadata, metadata_copy, append_list=False)
+
+    fluorescence_metadata = metadata_copy["Ophys"]["Fluorescence"]
+    fluorescence_name = fluorescence_metadata["name"]
+
+    ophys = get_module(nwbfile, "ophys")
+    if fluorescence_name in ophys.data_interfaces:
+        return nwbfile
+
+    add_plane_segmentation(segmentation_extractor=segmentation_extractor, nwbfile=nwbfile, metadata=metadata)
+
+    # Get plane segmentation from the image segmentation
+    image_segmentation_metadata = metadata_copy["Ophys"]["ImageSegmentation"]
+    image_segmentation_name = image_segmentation_metadata["name"]
+    image_segmentation = ophys.get_data_interface(image_segmentation_name)
+
+    plane_segmentation_name = image_segmentation_metadata["plane_segmentations"][0]["name"]
+    plane_segmentation = image_segmentation.plane_segmentations[plane_segmentation_name]
+
+    # TODO what should be done with plane_no_loop?
+    plane_no_loop = 0
```

Comment:
While we decided what to do with the multi segmentation extractor case `plane_no_loop` should be passed to the function. The `plane_no_loop` is used to determine the name of the objects and it varies with the outermost loop of `write_segmentation`.  So, to leave the behavior consistent with what we had before we should still name the objects with a variable `plane_no_loop` I think.

In the PR concerning add_summary_images I pass the name that already includes the effects of a variable `plane_no_loop` check that for a different approach as well.